### PR TITLE
Integration/db growth+recovery

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/invariants/consensus/NodeEvents.java [hunk 1]
```diff
@@ -39,6 +40,6 @@
 	}
 
 	public <T> EventProcessor<T> processor(BFTNode node, Class<T> eventClass) {
-		return t -> consumers.get(eventClass).forEach(c -> c.accept(node, t));
+		return t -> consumers.getOrDefault(eventClass, Collections.emptySet()).forEach(c -> c.accept(node, t));
```
Comment (siy) [inline radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/invariants/consensus/NodeEvents.java]: `Collections.emptySet()` can be replaced with `Set.of()` which is a little bit more concise.

File: radixdlt/src/integration/java/com/radixdlt/integration/recovery/RecoveryLivenessTest.java [hunk 2]
```diff
@@ -196,46 +207,149 @@ private void restartNode(int index) {
...
+		return network.allMessages().stream()
+			.filter(msg -> msg.message() instanceof EpochViewUpdate)
+			.map(msg -> (EpochViewUpdate) msg.message())
+			.map(e -> new EpochView(e.getEpoch(), e.getViewUpdate().getHighQC().highestCommittedQC().getView()))
+			.max(Comparator.naturalOrder());
+	}
+
+	private EpochView latestEpochView() {
+		return this.nodes.stream()
+			.map(i -> i.getInstance(Key.get(new TypeLiteral<DeterministicSavedLastEvent<EpochViewUpdate>>() { })).getLastEvent())
```
Comment (siy) [inline radixdlt/src/integration/java/com/radixdlt/integration/recovery/RecoveryLivenessTest.java]: The `Key.get(new TypeLiteral<DeterministicSavedLastEvent<EpochViewUpdate>>() { })` constant can be extracted.

File: radixdlt/src/integration/java/com/radixdlt/integration/recovery/RecoveryLivenessTest.java [hunk 3]
```diff
@@ -196,46 +207,149 @@ private void restartNode(int index) {
...
+
+		return senderIndex;
+	}
+
 	private void processForCount(int messageCount) {
 		for (int i = 0; i < messageCount; i++) {
-			Timed<ControlledMessage> msg = this.network.nextMessage();
-
-			Injector injector = this.nodes.get(msg.value().channelId().receiverIndex());
-			String bftNode = " " + injector.getInstance(Key.get(BFTNode.class, Self.class));
```
Comment (siy) [inline radixdlt/src/integration/java/com/radixdlt/integration/recovery/RecoveryLivenessTest.java:204]: The `Key.get(BFTNode.class, Self.class)` constant can be extracted.

File: radixdlt/src/main/java/com/radixdlt/DispatcherModule.java [hunk 4]
```diff
@@ -74,8 +78,12 @@ public void configure() {
 		Multibinder.newSetBinder(binder(), new TypeLiteral<EventProcessor<EpochViewUpdate>>() { }, ProcessOnDispatch.class);
 		Multibinder.newSetBinder(binder(), new TypeLiteral<EventProcessor<EpochViewUpdate>>() { });
 
+		Multibinder.newSetBinder(binder(), new TypeLiteral<EventProcessor<BFTInsertUpdate>>() { }, ProcessOnDispatch.class);
+		Multibinder.newSetBinder(binder(), new TypeLiteral<EventProcessor<BFTHighQCUpdate>>() { }, ProcessOnDispatch.class);
+		Multibinder.newSetBinder(binder(), new TypeLiteral<EventProcessor<BFTHighQCUpdate>>() { });
 		Multibinder.newSetBinder(binder(), new TypeLiteral<EventProcessor<BFTCommittedUpdate>>() { });
 		Multibinder.newSetBinder(binder(), new TypeLiteral<EventProcessor<BFTCommittedUpdate>>() { }, ProcessOnDispatch.class);
```
Comment (siy) [inline radixdlt/src/main/java/com/radixdlt/DispatcherModule.java]: Repeated type literals can be extracted as constants. This should improve readability.

File: radixdlt/src/main/java/com/radixdlt/store/berkeley/SerializedVertexStoreState.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import com.radixdlt.serialization.SerializerDummy;
+import com.radixdlt.serialization.SerializerId2;
+import java.util.Objects;
+
+/**
+ * Vertex Store State version which can be serialized.
+ */
+@SerializerId2("store.vertices")
+public class SerializedVertexStoreState {
+
```
Comment (siy) [inline radixdlt/src/main/java/com/radixdlt/store/berkeley/SerializedVertexStoreState.java:37]: I think such classes should have equals/hashCode/toString.

**Review by siy — APPROVED** [review]
LGTM

**Review by clstrfsck — APPROVED** [review]
Looks good, assume the test failure on merge is resolved.

**Pull request merged**